replibyte/src/config.rs

```diff
@@ -67,6 +68,15 @@ impl BridgeConfig {
     pub fn secret_access_key(&self) -> Result<String, Error> {
         substitute_env_var(self.secret_access_key.as_str())
     }
+
+    /// decode and return the endpoint value
+    pub fn endpoint(&self) -> Option<Result<String, Error>> {
```

Comment:
Here is my feedback:

1. I'd recommend changing the output signature for `Result<Option<String>, Error>`. It's not common to surround a Result<..> into an Option<..>.

2. A more "rust" way of doing would be to create an enum:

```
enum Endpoint {
  Custom { url: String },
  Default,
}
```

Then you can have the following signature:

```
pub fn endpoint(&self) -> Result<Endpoint, Error> {
  ...
}
```

WDYT?

Reply:
I think it will be much better ! I'll take a look to update the pull request later.

Reply:
I've added an Endpoint enum and remove the S3Builder because it no longer made sense. It's more readable like that. Let me know what you think.